includes/wikia/api/DWDimensionApiController.class.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+class DWDimensionApiController extends WikiaApiController {
+	const LIMIT = 100;
+	const AFTER_DOMAIN = null;
+
+	private function getSharedDbSlave() {
+		global $wgExternalSharedDB;
+		return wfGetDB( DB_SLAVE, array(), $wgExternalSharedDB );
+	}
+
+	public function getWikiDomains() {
+		$limit = $this->getRequest()->getVal( 'limit', static::LIMIT );
+		$afterDomain = $this->getRequest()->getVal( 'after_domain', static::AFTER_DOMAIN );
```

Comment:
`after_domain`?

How will this API be called?

Reply:
first call you don't specify anything and then you take the last value of domain you received and set it in the after_domain params - you'll get everything than comes after.

I was researching how to best approach pagination and I saw limit supports setting two parameters, but read that for large data sets a where clause is more efficient as it doesn't need to get all the entries and then filter. Also I saw this table has an index on domain column and city_id unfortunatelly is not unique, so cannot be used.

Does that make sense or do you have a better idea?